### PR TITLE
Remove not used SimplePool

### PR DESCRIPTION
**Pull request by MishaDemianenko**

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java [hunk 1]
```diff
@@ -100,9 +101,11 @@ public BatchingNeoStores( FileSystemAbstraction fileSystem, File storeDir, Confi
         // Having less than that might result in an evicted page will reading, which would mean
         // unnecessary re-reading. Having slightly more leaves some leg room.
         long optimalMappedMemorySize = pageSize * 40;
+        // page cache requires at the very least memory enough for two pages
+        long limitedMemorySize = max( 2 * pageSize,  applyEnvironmentLimitationsTo( optimalMappedMemorySize ) );
         this.neo4jConfig = dbConfig.with( stringMap(
                 dense_node_threshold.name(), valueOf( config.denseNodeThreshold() ),
-                pagecache_memory.name(), valueOf( applyEnvironmentLimitationsTo( optimalMappedMemorySize ) ),
+                pagecache_memory.name(), valueOf( limitedMemorySize ),
                 mapped_memory_page_size.name(), valueOf( pageSize ) ),
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java]: Looks unrelated to this PR, and will conflict with #6645

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java]: absolutely true, but needs to be here to be able to pass tests

**Comment by chrisvest** [comment]
@MishaDemianenko I merged #6645, so you can rebase and clean it up.

**Comment by MishaDemianenko** [comment]
@chrisvest rebased and clean
